ResearchKit/ActiveTasks/ORKLocationRecorder.m

```diff
@@ -116,36 +112,74 @@ - (void)doStopRecording {
 
 - (void)stop {
     [self doStopRecording];
-    [_logger finishCurrentLog];
-    
-    NSError *error = _recordingError;
-    _recordingError = nil;
-    __block NSURL *fileUrl = nil;
-    [_logger enumerateLogs:^(NSURL *logFileUrl, BOOL *stop) {
-        fileUrl = logFileUrl;
-    } error:&error];
-    
-    [self reportFileResultWithFile:fileUrl error:error];
     
     [super stop];
 }
 
 - (void)locationManager:(CLLocationManager *)manager
      didUpdateLocations:(NSArray *)locations {
     BOOL success = YES;
-    BOOL relativeDistanceOnly = ((ORKLocationRecorderConfiguration *)self.configuration).relativeDistanceOnly;
+    BOOL relativeDistanceOnly = [self relativeDistanceOnly];
     NSParameterAssert(locations.count >= 0);
     NSError *error = nil;
     if (locations) {
-        NSMutableArray *dictionaries = [NSMutableArray arrayWithCapacity:locations.count];
-        [locations enumerateObjectsUsingBlock:^(CLLocation *obj, NSUInteger idx, BOOL *stop) {
-            NSDictionary *d = [obj ork_jsonCodingObjectWithRelativeDistanceOnly:relativeDistanceOnly
-                                                                       previous:_lastLocation];
-            [dictionaries addObject:d];
-            _lastLocation = [obj copy];
-        }];
         
-        success = [_logger appendObjects:dictionaries error:&error];
+        // Add the location data points to the log
+        __block NSMutableArray *dictionaries = [NSMutableArray arrayWithCapacity:locations.count];
+        dispatch_sync(_processingQueue, ^{
+            [locations enumerateObjectsUsingBlock:^(CLLocation *obj, NSUInteger idx, BOOL *stop) {
+                
+                // Calculate time interval since start time
+                NSTimeInterval timeInterval = [obj.timestamp timeIntervalSinceDate:self.startDate];
+                
+                // If there is a reference uptime or the data is consolidated then this uses a schema that
+                // uses an NSTimeInterval to define the timestamp rather than a date. This remains compatible
+                // with the V1 schema, while allowing the data to be compared to other recorders using the
+                // same value for the timestamp.
+                id timestamp = nil;
+                BOOL consolidated = (self.sharedLogger != nil);
+                if (consolidated || (self.referenceUptime > 0)) {
+                    NSTimeInterval uptimeDelta = (self.referenceUptime > 0) ? (self.uptime - self.referenceUptime) : 0;
+                    timestamp = [NSDecimalNumber numberWithDouble:(uptimeDelta + timeInterval)];
+                }
+                
+                // Only include total distance traveled if the user is not *suppose* to be standing still
+                // and the horizontal accuracy indicates that the user is outdoors
+                if (!self.isStandingStill && (obj.horizontalAccuracy > 0) && (obj.horizontalAccuracy <= ORKLocationRequiredAccuracy)) {
```

Comment:
Again, it's a really rough guide right now. I *might* look at improving the accuracy and brush up on my geometry but that's a project for the future.